docs/source/python/recipe/postgresql_pandas.py

```diff
@@ -0,0 +1,65 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# RECIPE STARTS HERE
+
+#: ADBC is integrated into Pandas_, a popular dataframe library.  Pandas can
+#: use ADBC to read tables in PostgreSQL databases.  Compared to using
+#: SQLAlchemy or other options, using ADBC with Pandas can have better
+#: performance, such as by avoiding excess conversions to and from Python
+#: objects.
+#:
+#: .. _Pandas: https://pandas.pydata.org/
+
+import os
+
+import pandas
+import pyarrow
+
+import adbc_driver_postgresql.dbapi
+
+uri = os.environ["ADBC_POSTGRESQL_TEST_URI"]
+conn = adbc_driver_postgresql.dbapi.connect(uri)
+
+#: For the purposes of testing, we'll first make sure the tables we're about
+#: to use don't exist.
+with conn.cursor() as cur:
+    cur.execute("DROP TABLE IF EXISTS example")
+
+#: Then we'll use ADBC to create a sample table.  (ADBC is not currently
```

Comment:
You should be able to use this with `pd.to_sql`

```python
df.to_sql("example", conn, index=False, if_exists="replace")
```

It does require a minimum version of pandas 2.2

https://pandas.pydata.org/pandas-docs/stable/whatsnew/v2.2.0.html#adbc-driver-support-in-to-sql-and-read-sql

Reply:
Ah, thanks - I'll update this

Reply:
Ah, I think I got confused since the docstring itself doesn't mention ADBC - https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_sql.html#pandas.DataFrame.to_sql

Reply:
That's an unfortunate oversight - will get that corrected

https://github.com/pandas-dev/pandas/issues/59095